### PR TITLE
#AV-63190 Updated module fields and added type for each field in docu…

### PR DESCRIPTION
**Pull request by Rohan-sss1**
…mentation

##### SUMMARY

Updated following modules for Avi version 18.2.8
1. avi_actiongroupconfig.py
2. avi_alertconfig.py
3. avi_alertemailconfig.py
4. avi_alertscriptconfig.py
5. avi_alertsyslogconfig.py
6. avi_analyticsprofile.py
7. avi_api_session.py
8. avi_api_version.py
9. avi_applicationpersistenceprofile.py
10. avi_applicationprofile.py
11. avi_authprofile.py
12. avi_autoscalelaunchconfig.py
13. avi_backup.py
14. avi_backupconfiguration.py
15. avi_certificatemanagementprofile.py
16. avi_cloud.py
17. avi_cloudconnectoruser.py
18. avi_cloudproperties.py
19. avi_cluster.py
20. avi_clusterclouddetails.py
21. avi_controllerproperties.py
22. avi_customipamdnsprofile.py
23. avi_dnspolicy.py
24. avi_errorpagebody.py
25. avi_errorpageprofile.py
26. avi_gslb.py
27. avi_gslbgeodbprofile.py
28. avi_gslbservice.py
29. avi_gslbservice_patch_member.py
30. avi_hardwaresecuritymodulegroup.py
31. avi_healthmonitor.py
32. avi_httppolicyset.py
33. avi_ipaddrgroup.py
34. avi_ipamdnsproviderprofile.py
35. avi_l4policyset.py
36. avi_microservicegroup.py
37. avi_network.py
38. avi_networkprofile.py
39. avi_networksecuritypolicy.py
40. avi_pkiprofile.py
41. avi_pool.py
42. avi_poolgroup.py
43. avi_poolgroupdeploymentpolicy.py
44. avi_prioritylabels.py
45. avi_role.py
46. avi_scheduler.py
47. avi_seproperties.py
48. avi_serverautoscalepolicy.py
49. avi_serviceengine.py
50. avi_serviceenginegroup.py
51. avi_snmptrapprofile.py
52. avi_sslkeyandcertificate.py
53. avi_sslprofile.py
54. avi_stringgroup.py
55. avi_systemconfiguration.py
56. avi_tenant.py
57. avi_trafficcloneprofile.py
58. avi_user.py
59. avi_useraccount.py
60. avi_useraccountprofile.py
61. avi_virtualservice.py
62. avi_vrfcontext.py
63. avi_vsdatascriptset.py
64. avi_vsvip.py
65. avi_webhook.py

##### ISSUE TYPE

- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME

network/avi/avi_actiongroupconfig.py
network/avi/avi_alertconfig.py
network/avi/avi_alertemailconfig.py
network/avi/avi_alertscriptconfig.py
network/avi/avi_alertsyslogconfig.py
network/avi/avi_analyticsprofile.py
network/avi/avi_api_session.py
network/avi/avi_api_version.py
network/avi/avi_applicationpersistenceprofile.py
network/avi/avi_applicationprofile.py
network/avi/avi_authprofile.py
network/avi/avi_autoscalelaunchconfig.py
network/avi/avi_backup.py
network/avi/avi_backupconfiguration.py
network/avi/avi_certificatemanagementprofile.py
network/avi/avi_cloud.py
network/avi/avi_cloudconnectoruser.py
network/avi/avi_cloudproperties.py
network/avi/avi_cluster.py
network/avi/avi_clusterclouddetails.py
network/avi/avi_controllerproperties.py
network/avi/avi_customipamdnsprofile.py
network/avi/avi_dnspolicy.py
network/avi/avi_errorpagebody.py
network/avi/avi_errorpageprofile.py
network/avi/avi_gslb.py
network/avi/avi_gslbgeodbprofile.py
network/avi/avi_gslbservice.py
network/avi/avi_gslbservice_patch_member.py
network/avi/avi_hardwaresecuritymodulegroup.py
network/avi/avi_healthmonitor.py
network/avi/avi_httppolicyset.py
network/avi/avi_ipaddrgroup.py
network/avi/avi_ipamdnsproviderprofile.py
network/avi/avi_l4policyset.py
network/avi/avi_microservicegroup.py
network/avi/avi_network.py
network/avi/avi_networkprofile.py
network/avi/avi_networksecuritypolicy.py
network/avi/avi_pkiprofile.py
network/avi/avi_pool.py
network/avi/avi_poolgroup.py
network/avi/avi_poolgroupdeploymentpolicy.py
network/avi/avi_prioritylabels.py
network/avi/avi_role.py
network/avi/avi_scheduler.py
network/avi/avi_seproperties.py
network/avi/avi_serverautoscalepolicy.py
network/avi/avi_serviceengine.py
network/avi/avi_serviceenginegroup.py
network/avi/avi_snmptrapprofile.py
network/avi/avi_sslkeyandcertificate.py
network/avi/avi_sslprofile.py
network/avi/avi_stringgroup.py
network/avi/avi_systemconfiguration.py
network/avi/avi_tenant.py
network/avi/avi_trafficcloneprofile.py
network/avi/avi_user.py
network/avi/avi_useraccount.py
network/avi/avi_useraccountprofile.py
network/avi/avi_virtualservice.py
network/avi/avi_vrfcontext.py
network/avi/avi_vsdatascriptset.py
network/avi/avi_vsvip.py
network/avi/avi_webhook.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/network/avi/avi_alertconfig.py [hunk 1]
```diff
@@ -74,89 +81,70 @@
...
         description:
             - Uuid of the resource for which alert was raised.
+        type: str
     object_type:
         description:
             - The object type to which the alert config is associated with.
             - Valid object types are - virtual service, pool, service engine.
             - Enum options - VIRTUALSERVICE, POOL, HEALTHMONITOR, NETWORKPROFILE, APPLICATIONPROFILE, HTTPPOLICYSET, DNSPOLICY, SECURITYPOLICY, IPADDRGROUP,
             - STRINGGROUP, SSLPROFILE, SSLKEYANDCERTIFICATE, NETWORKSECURITYPOLICY, APPLICATIONPERSISTENCEPROFILE, ANALYTICSPROFILE, VSDATASCRIPTSET, TENANT,
-            - PKIPROFILE, AUTHPROFILE, CLOUD, SERVERAUTOSCALEPOLICY, AUTOSCALELAUNCHCONFIG, MICROSERVICEGROUP, IPAMPROFILE, HARDWARESECURITYMODULEGROUP,
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_alertconfig.py:90]: Wow, this description is really not very helpful... Is there a documentation that could be linked instead of removing all these entries?

If not, it's probably better to keep them, but format them nicer. Like using `C(...)` for every value, and not using new paragraphs but having them all in one paragraph (i.e. remove `-` at beginning of line).

File: lib/ansible/modules/network/avi/avi_analyticsprofile.py [hunk 2]
```diff
@@ -276,156 +302,196 @@
...
             - Allowed values are 0-5.
             - Default value when not specified in API or module is interpreted by Avi Controller as 5.0.
+        type: float
+    hs_security_tls13_score:
+        description:
+            - Score assigned when supporting tls1.3 encryption protocol.
+            - Allowed values are 0-5.
+            - Field introduced in 18.2.6.
+            - Default value when not specified in API or module is interpreted by Avi Controller as 5.0.
+        version_added: "2.10"
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_analyticsprofile.py:459]: Please do not add new options in such a big bugfix PR. Otherwise, you'll never get this properly reviewed and merged.

File: lib/ansible/modules/network/avi/avi_api_session.py [hunk 3]
```diff
@@ -168,7 +189,8 @@ def main():
     gparams.update({'include_refs': '', 'include_name': ''})
 
     # API methods not allowed
-    api_get_not_allowed = ["cluster", "gslbsiteops"]
+    api_get_not_allowed = ["cluster", "gslbsiteops", "server"]
+    sub_api_get_not_allowed = ["scaleout"]
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_api_session.py:193]: Please put these changes into another PR. Make one big PR for adding types, and multiple other PRs for other changes. (Also, the other PRs should have [changelog fragments](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).)

File: lib/ansible/modules/network/avi/avi_cloud.py [hunk 4]
```diff
@@ -30,47 +30,57 @@
...
     apic_mode:
         description:
             - Boolean flag to set apic_mode.
             - Default value when not specified in API or module is interpreted by Avi Controller as False.
         type: bool
     autoscale_polling_interval:
         description:
-            - Cloudconnector polling interval for external autoscale groups.
+            - Cloudconnector polling interval in seconds for external autoscale groups, minimum 60 seconds.
+            - Allowed values are 60-3600.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_cloud.py:60]: Such changes might also better go into other PRs, since these need to be reviewed by someone who knows what these modules are actually doing ;-)

File: lib/ansible/modules/network/avi/avi_pool.py [hunk 5]
```diff
@@ -30,61 +30,74 @@
...
     application_persistence_profile_ref:
         description:
             - Persistence will ensure the same user sticks to the same server for a desired duration of time.
             - It is a reference to an object of type applicationpersistenceprofile.
+        type: str
     autoscale_launch_config_ref:
         description:
             - If configured then avi will trigger orchestration of pool server creation and deletion.
-            - It is only supported for container clouds like mesos, openshift, kubernetes, docker, etc.
+            - It is only supported for container clouds like mesos, opensift, kubernates, docker etc.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_pool.py:89]:
```suggestion
            - It is only supported for container clouds like mesos, openshift, kubernates, docker etc.
```
The `h` was supposed to be there.

File: lib/ansible/modules/network/avi/avi_pool.py [hunk 6]
```diff
@@ -96,34 +109,41 @@
...
             - Internally set by cloud connector.
+        type: str
     cloud_ref:
         description:
             - It is a reference to an object of type cloud.
+        type: str
     conn_pool_properties:
         description:
-            - Connection pool properties.
+            - Connnection pool properties.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_pool.py:124]:
```suggestion
            - Connection pool properties.
```

File: lib/ansible/modules/network/avi/avi_pool.py [hunk 7]
```diff
@@ -201,30 +228,35 @@
...
             - Default value when not specified in API or module is interpreted by Avi Controller as LB_ALGORITHM_LEAST_CONNECTIONS.
+        type: str
     lb_algorithm_consistent_hash_hdr:
         description:
             - Http header name to be used for the hash key.
+        type: str
     lb_algorithm_core_nonaffinity:
         description:
-            - Degree of non-affinity for core affinity based server selection.
+            - Degree of non-affinity for core afffinity based server selection.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_pool.py:246]:
```suggestion
            - Degree of non-affinity for core affinity based server selection.
```

File: lib/ansible/modules/network/avi/avi_pool.py [hunk 8]
```diff
@@ -363,12 +416,16 @@
     uuid:
         description:
             - Uuid of the pool.
+        type: str
     vrf_ref:
         description:
             - Virtual routing context that the pool is bound to.
             - This is used to provide the isolation of the set of networks the pool is attached to.
-            - The pool inherits the virtual routing context of the virtual service, and this field is used only internally, and is set by pb-transform.
+            - The pool inherits the virtual routing conext of the virtual service, and this field is used only internally, and is set by pb-transform.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_pool.py:424]:
```suggestion
            - The pool inherits the virtual routing context of the virtual service, and this field is used only internally, and is set by pb-transform.
```

File: lib/ansible/modules/network/avi/avi_serviceengine.py [hunk 9]
```diff
@@ -29,33 +29,39 @@
...
+        type: str
     container_mode:
         description:
             - Boolean flag to set container_mode.
             - Default value when not specified in API or module is interpreted by Avi Controller as False.
         type: bool
     container_type:
         description:
-            - Enum options - container_type_bridge, container_type_host, container_type_host_dpdk.
+            - Enum options - CONTAINER_TYPE_BRIDGE, CONTAINER_TYPE_HOST, CONTAINER_TYPE_HOST_DPDK.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_serviceengine.py:62]: Do you have some docs links that this is really the case? Or (better) also move this into another PR.

File: lib/ansible/modules/network/avi/avi_serviceenginegroup.py [hunk 10]
```diff
@@ -62,10 +65,11 @@
             - Field deprecated in 18.1.2.
             - Field introduced in 18.1.1.
         version_added: "2.9"
+        type: int
     advertise_backend_networks:
         description:
             - Advertise reach-ability of backend server networks via adc through bgp for default gateway feature.
-            - Default value when not specified in API or module is interpreted by Avi Controller as False.
+            - Field deprecated in 18.2.5.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_serviceenginegroup.py:72]: Same here.

File: lib/ansible/modules/network/avi/avi_serviceenginegroup.py [hunk 11]
```diff
@@ -85,13 +90,14 @@
         type: bool
     app_cache_percent:
         description:
-            - A percent value of total se memory reserved for application caching.
-            - This is an se bootup property and requires se restart.
+            - A percent value of total se memory reserved for applicationcaching.
+            - This is an se bootup property and requires se restart.requires se reboot.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_serviceenginegroup.py:94]: Please remove the changes from the above two lines.

File: lib/ansible/modules/network/avi/avi_serviceenginegroup.py [hunk 12]
```diff
@@ -100,23 +106,26 @@
...
     archive_shm_limit:
         description:
             - Amount of se memory in gb until which shared memory is collected in core archive.
             - Field introduced in 17.1.3.
             - Default value when not specified in API or module is interpreted by Avi Controller as 8.
+        type: int
     async_ssl:
         description:
-            - Ssl handshakes will be handled by dedicated ssl threads.
+            - Ssl handshakes will be handled by dedicated ssl threads.requires se reboot.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_serviceenginegroup.py:118]:
```suggestion
            - Ssl handshakes will be handled by dedicated ssl threads.requires se reboot.
```
This is not correct English.

File: lib/ansible/modules/network/avi/avi_serviceenginegroup.py [hunk 13]
```diff
@@ -149,15 +161,18 @@
             - Bgp peer state update interval.
             - Allowed values are 5-100.
             - Field introduced in 17.2.14,18.1.5,18.2.1.
-            - Default value when not specified in API or module is interpreted by Avi Controller as 10.
+            - Default value when not specified in API or module is interpreted by Avi Controller as 60.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_serviceenginegroup.py:164]: Similar as above.

File: lib/ansible/modules/network/avi/avi_serviceenginegroup.py [hunk 14]
```diff
@@ -216,12 +237,13 @@
...
         description:
             - User defined description for the object.
+        type: str
     disable_avi_securitygroups:
         description:
             - By default, avi creates and manages security groups along with custom sg provided by user.
             - Set this to true to disallow avi to create and manage new security groups.
             - Avi will only make use of custom security groups provided by user.
-            - This option is only supported for aws cloud type.
+            - This option is supported for aws and openstack cloud types.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_serviceenginegroup.py:246]: Same as above.

File: lib/ansible/modules/network/avi/avi_virtualservice.py [hunk 15]
```diff
@@ -273,52 +307,64 @@
...
     nsx_securitygroup:
         description:
             - A list of nsx service groups representing the clients which can access the virtual ip of the virtual service.
             - Field introduced in 17.1.1.
         version_added: "2.4"
+        type: list
     performance_limits:
         description:
-            - Optional settings that determine performance limits like max connections or bandwidth etc.
+            - Optional settings that determine performance limits like max connections or bandwdith etc.
```
Comment (felixfontein) [inline lib/ansible/modules/network/avi/avi_virtualservice.py:351]: What's the change here? A different newline type maybe?

**Comment by Akasurde** [comment]
Hi @Rohan-sss1, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.network<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>
